localstack/utils/container_utils/docker_cmd_client.py

```diff
@@ -408,6 +409,39 @@ def stream_container_logs(self, container_name_or_id: str) -> CancellableStream:
 
         return CancellableProcessStream(process)
 
+    def events(
+        self,
+        since: Optional[Union[datetime, int]] = None,
+        until: Optional[Union[datetime, int]] = None,
+        filters: Optional[Dict] = None,
+    ) -> Generator[Dict, None, None]:
+        cmd = self._docker_cmd()
+        cmd += ["events", "--format", "{{json .}}"]
+
+        def _to_timestamp(value: Union[datetime, int]):
+            if isinstance(value, datetime):
+                return value.timestamp()
+            else:
+                return value
+
+        if since:
+            cmd += [_to_timestamp(since)]
+        if until:
+            cmd += ["--until", _to_timestamp(until)]
+        if filters:
+            filter_pairs = [f"{key}={value}" for (key, value) in filters.items()]
+            cmd += ["--filter", ",".join(filter_pairs)]
+
+        process: subprocess.Popen = run(
+            cmd, asynchronous=True, outfile=subprocess.PIPE, stderr=subprocess.STDOUT
+        )
+
+        for msg in CancellableProcessStream(process):
+            try:
+                yield json.loads(msg)
+            except json.JSONDecodeError as e:
+                LOG.warning("Error decoding docker event %s: %s", msg, e)
+
```

Comment:
Not sure about this. The stream is not cancellable with this construct. I think for these method we should always use a CancellableStream, so we can cancel it if necessary.